codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyFrameCodec.java

```diff
@@ -125,13 +125,18 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
 
     @Override
     public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
-        if (!read) {
+        boolean wasRead = read;
+        read = false;
+        if (!wasRead) {
             if (!ctx.channel().config().isAutoRead()) {
                 ctx.read();
             }
         }
-        read = false;
         super.channelReadComplete(ctx);
+
+        if (wasRead) {
```

Comment:
`super.channelReadComplete(ctx);` may also call `ctx.fireChannelReadComplete();` ... are we sure this method won't be called twice?